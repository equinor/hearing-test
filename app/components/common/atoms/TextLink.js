import React from 'react';
import PropTypes from 'prop-types';
import { Text, View, StyleSheet, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';
import * as Colors from '../../../stylesheets/colors';

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  linkContainer: {
    flex: 1,
    flexDirection: 'row',
  },
  defaultText: {
    flex: 1,
  },
  icon: {
    justifyContent: 'flex-end',
    alignSelf: 'center',
  },
});

const TextLink = ({ data, textStyle, nav }) => (
  <TouchableOpacity
    onPress={() => nav.navigate(data.route)}
    underlayColor="#00000000"
    style={styles.container}
  >
    <View style={styles.linkContainer}>
      <Text style={[textStyle, styles.defaultText]}>{data.name}</Text>
      <Icon name="ios-arrow-forward" style={styles.icon} size={20} color={Colors.GRAY_2} />
    </View>
  </TouchableOpacity>
);

TextLink.propTypes = {
  data: PropTypes.shape({
    name: PropTypes.string.isRequired,
    route: PropTypes.string,
  }).isRequired,
  textStyle: Text.propTypes.style, // eslint-disable-line
  nav: PropTypes.shape({
    state: PropTypes.object,
    navigate: PropTypes.func,
  }).isRequired,
};

TextLink.defaultProps = {
  textStyle: {},
};

export default TextLink;
