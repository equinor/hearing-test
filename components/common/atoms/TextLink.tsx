import { MaterialIcons as Icon } from "@expo/vector-icons";
import React from "react";
import {
  StyleProp,
  StyleSheet,
  Text,
  TextStyle,
  TouchableOpacity,
  View,
} from "react-native";

import * as Colors from "../../../constants/colors";

const TextLink = ({
  data,
  textStyle = {},
  nav,
}: {
  data: { name: string; route?: string };
  textStyle: StyleProp<TextStyle>;
  nav: { state: object; navigate: Function };
}) => (
  <TouchableOpacity
    onPress={() => nav.navigate(data.route)}
    style={styles.container}
  >
    <View style={styles.linkContainer}>
      <Text style={[styles.defaultText, textStyle]}>{data.name}</Text>
      <Icon
        color={Colors.GRAY_2}
        name="arrow-forward-ios"
        size={12}
        style={styles.icon}
      />
    </View>
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  defaultText: {
    flex: 1,
  },
  icon: {
    alignSelf: "center",
    justifyContent: "flex-end",
  },
  linkContainer: {
    flex: 1,
    flexDirection: "row",
  },
});

export default TextLink;
