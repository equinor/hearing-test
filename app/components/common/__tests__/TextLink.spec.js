import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import sinon from 'sinon';
import TextLink from '../atoms/TextLink';

it('renders and behaves correctly', () => {
  let returnedRoute;
  const nav = {
    navigate: sinon.spy(route => {
      returnedRoute = route;
    }),
  };
  const wrapper = shallow(<TextLink data={{ name: 'test', route: 'testRoute' }} nav={nav} />);
  const TextLinkControl = wrapper.find('TouchableOpacity');
  expect(wrapper).toMatchSnapshot();
  expect(nav.navigate.calledOnce).toBeFalsy();
  TextLinkControl.simulate('press');
  expect(nav.navigate.calledOnce).toBeTruthy();
  expect(returnedRoute).toBe('testRoute');
});
