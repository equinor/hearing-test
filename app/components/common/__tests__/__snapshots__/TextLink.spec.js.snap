// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders and behaves correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TextLink
    data={
      Object {
        "name": "test",
        "route": "testRoute",
      }
    }
    nav={
      Object {
        "navigate": [Function],
      }
    }
    textStyle={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "activeOpacity": 0.2,
      "children": <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "row",
          }
        }
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              Object {
                "flex": 1,
              },
            ]
          }
        >
          test
        </Text>
        <Icon
          allowFontScaling={false}
          color="#808080"
          name="ios-arrow-forward"
          size={20}
          style={
            Object {
              "alignSelf": "center",
              "justifyContent": "flex-end",
            }
          }
        />
      </View>,
      "onPress": [Function],
      "style": Object {
        "flex": 1,
      },
      "underlayColor": "#00000000",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            test
          </Text>,
          <Icon
            allowFontScaling={false}
            color="#808080"
            name="ios-arrow-forward"
            size={20}
            style={
              Object {
                "alignSelf": "center",
                "justifyContent": "flex-end",
              }
            }
          />,
        ],
        "style": Object {
          "flex": 1,
          "flexDirection": "row",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "accessible": true,
            "allowFontScaling": true,
            "children": "test",
            "ellipsizeMode": "tail",
            "style": Array [
              Object {},
              Object {
                "flex": 1,
              },
            ],
          },
          "ref": null,
          "rendered": "test",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "allowFontScaling": false,
            "color": "#808080",
            "name": "ios-arrow-forward",
            "size": 20,
            "style": Object {
              "alignSelf": "center",
              "justifyContent": "flex-end",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "activeOpacity": 0.2,
        "children": <View
          style={
            Object {
              "flex": 1,
              "flexDirection": "row",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {},
                Object {
                  "flex": 1,
                },
              ]
            }
          >
            test
          </Text>
          <Icon
            allowFontScaling={false}
            color="#808080"
            name="ios-arrow-forward"
            size={20}
            style={
              Object {
                "alignSelf": "center",
                "justifyContent": "flex-end",
              }
            }
          />
        </View>,
        "onPress": [Function],
        "style": Object {
          "flex": 1,
        },
        "underlayColor": "#00000000",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {},
                  Object {
                    "flex": 1,
                  },
                ]
              }
            >
              test
            </Text>,
            <Icon
              allowFontScaling={false}
              color="#808080"
              name="ios-arrow-forward"
              size={20}
              style={
                Object {
                  "alignSelf": "center",
                  "justifyContent": "flex-end",
                }
              }
            />,
          ],
          "style": Object {
            "flex": 1,
            "flexDirection": "row",
          },
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "accessible": true,
              "allowFontScaling": true,
              "children": "test",
              "ellipsizeMode": "tail",
              "style": Array [
                Object {},
                Object {
                  "flex": 1,
                },
              ],
            },
            "ref": null,
            "rendered": "test",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "allowFontScaling": false,
              "color": "#808080",
              "name": "ios-arrow-forward",
              "size": 20,
              "style": Object {
                "alignSelf": "center",
                "justifyContent": "flex-end",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
